components/camera/camera.go

```diff
@@ -135,7 +135,13 @@ func NewFromReader(
 			if err != nil {
 				return nil, NewPropertiesError("source camera")
 			}
-			actualSystem = &transform.PinholeCameraModel{props.IntrinsicParams, props.DistortionParams}
+
+			var cameraModel transform.PinholeCameraModel
```

Comment:
This fixed the issue in slam. The rest of the changes are to ensure that the issue doesn't surface anywhere else.